bentoml/_internal/bento/gen.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+import re
+import typing as t
+import logging
+from typing import TYPE_CHECKING
+
+import fs
+from jinja2 import Environment
+from jinja2.loaders import FileSystemLoader
+
+from ..utils import bentoml_cattr
+from .docker import DistroSpec
+from ...exceptions import BentoMLException
+from ..configuration import BENTOML_VERSION
+
+logger = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+    P = t.ParamSpec("P")
+    from .build_config import DockerOptions
+
+    TemplateFunc = t.Callable[[DockerOptions], t.Dict[str, t.Any]]
+
+
+def clean_bentoml_version() -> str:
+    post_version = BENTOML_VERSION.split("+")[0]
+    match = re.match(r"^(\d+).(\d+).(\d+)(?:(a|rc)\d)", post_version)
+    if match is None:
+        raise BentoMLException("Errors while parsing BentoML version.")
+    return match.group()
+
+
+DEFAULT_UID_GID = 1034
+DEFAULT_BENTO_USER = "bentoml"
+DEFAULT_BENTO_HOME = f"/home/{DEFAULT_BENTO_USER}"
+DEFAULT_BENTO_PATH = f"{DEFAULT_BENTO_HOME}/bento"
+
+
+def get_template_env(docker_options: DockerOptions) -> dict[str, t.Any]:
+    distro = docker_options.distro
+    cuda_version = docker_options.cuda_version
+    python_version = docker_options.python_version
+
+    distro_spec = DistroSpec.from_distro(distro, cuda=cuda_version not in (None, ""))
+
+    if docker_options.base_image is None:
+        if cuda_version not in ("", None):
+            base_image = distro_spec.image.format(cuda_version=cuda_version)
+        else:
+            if distro in ["ubi8"]:
+                python_version = python_version.replace(".", "")
+            else:
+                python_version = python_version
+        base_image = distro_spec.image.format(python_version=python_version)
+    else:
+        base_image = docker_options.base_image
+        logger.warning(f"Make sure to have Python installed for {base_image}.")
+
+    return {
+        "base_image": base_image,
+        "bentoml_version": clean_bentoml_version(),
+        "default_uid_gid": DEFAULT_UID_GID,
```

Comment:
Is there a reason these are called `default...`? I think it makes more sense to not have the default there unless it's actually configurable, and even then the template variables probably shouldn't have that.

Reply:
I was thinking to use this for configurable variable.
so for the blocks users can use `SETUP_BENTO_<>` block
for variable we will have a few with `bento__<variable>` format
Let me remove the default tag then